concourse/scripts/spec/script_testing_helpers.rb

```diff
@@ -0,0 +1,100 @@
+require "aruba/rspec"
+require "pathname"
+
+def find_git_repository_path()
+  path = __FILE__
+  begin
+    if File.exists? File.join(path, '.git')
+      return path
+    end
+    path = File.realpath(File.join(path, '..'))
+  end while path != "/"
```

Comment:
I found this quite difficult to read. I think `git rev-parse --show-toplevel` could do the same thing.